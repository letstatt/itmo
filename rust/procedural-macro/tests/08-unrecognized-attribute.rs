// Этот тест проверяет, что пользовать получить человекочитаемое сообщение об ошибке в случае
// опечатки в названии атрибута.
//
// Предпочтительный способ генерации ошибок – генерация код с макросом `compile_error!`.
//
// Ссылки:
//
//   - compile_error! макрос:
//     https://doc.rust-lang.org/std/macro.compile_error.html
//
//   - Преобразование `syn::Error` в вызов `compile_error!`:
//     https://docs.rs/syn/1.0/syn/struct.Error.html#method.to_compile_error

use derive_builder::Builder;

#[derive(Builder)]
pub struct Command {
    executable: String,
    #[builder(eac = "arg")]
    args: Vec<String>,
    env: Vec<String>,
    current_dir: Option<String>,
}

fn main() {}
