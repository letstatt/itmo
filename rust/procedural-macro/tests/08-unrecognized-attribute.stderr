error: expected `builder(each = "...")`
  --> tests/08-unrecognized-attribute.rs:19:7
   |
19 |     #[builder(eac = "arg")]
   |       ^^^^^^^^^^^^^^^^^^^^
